pkgs/by-name/sl/slimevr/default.nix

```diff
@@ -0,0 +1,221 @@
+      ./no-grgit.patch
+    ];
+
+    postPatch = ''
+      # Manually apply program version, since Git can't be accessed.
+      substituteInPlace server/desktop/build.gradle.kts \
+        --replace-fail 'val gitVersionTag = ""' 'val gitVersionTag = "v${version}"'
+
+      # Disable Android, so its files don't have to be patched.
+      substituteInPlace settings.gradle.kts \
+        --replace-fail 'include(":server:android")' ""
+    '';
+
+    nativeBuildInputs = [
+      gradle
+      makeWrapper
+    ];
+
+    # this is required for using mitm-cache on Darwin
+    __darwinAllowLocalNetworking = true;
+
+    gradleFlags = [ "-Dorg.gradle.java.home=${jdk17_headless}" ];
+
+    gradleBuildTask = "shadowJar";
+
+    doCheck = true;
+
+    installPhase = ''
+      runHook preInstall
+
+      install -Dm644 server/desktop/build/libs/slimevr.jar $out/share/slimevr/slimevr.jar
+      makeWrapper ${jdk17_headless}/bin/java $out/bin/slimevr-server \
+          --add-flags "${serverJavaOptions}" \
+          --add-flags "-jar $out/share/slimevr/slimevr.jar"
+
+      runHook postInstall
+    '';
+
+    meta = sharedMeta // {
+      mainProgram = "slimevr-server";
+    };
+  });
+in
+{
+  slimevr-server = server;
+  slimevr = rustPlatform.buildRustPackage {
+    inherit version pname src;
+
+    # `hyper-tls` and `rustls-pemfile` aren't in the upstream lockfile, but are in
+    # Tauri's. This lock file is the upstream one with both of those added.
+    # `hper-tls` and `rustls-pemfile` are optional dependencies of `reqwest`.
+    cargoLock = {
+      lockFile = ./Cargo.lock;
+    };
+
+    pnpmDeps = pnpm_9.fetchDeps {
+      pname = "${pname}-pnpm-deps";
+      inherit version src;
+      hash = "sha256-5IqIUwVvufrws6/xpCAilmgRNG4mUGX8NXajZcVZypM=";
+    };
+
+    nativeBuildInputs = [
+      nodejs
+      pnpm_9.configHook
+      cargo-tauri.hook
+      pkg-config
+      wrapGAppsHook3
+      makeWrapper
+    ];
+
+    buildInputs =
+      [
+        openssl.dev
+        gst_all_1.gstreamer
+        gst_all_1.gst-plugins-base
+        gst_all_1.gst-plugins-good
+        gst_all_1.gst-plugins-bad
+      ]
+      ++ lib.optionals stdenv.isLinux [
+        glib-networking
+        libayatana-appindicator
+        webkitgtk_4_1
+      ];
+
+    postPatch =
+      ''
+        rm Cargo.lock
+        cp --no-preserve=mode ${./Cargo.lock} Cargo.lock
+
+        # Manually apply program version in place of Git autodetection.
+        sed '/git --no-pager tag /{n;N;N;d}' -i gui/vite.config.ts
+        substituteInPlace gui/vite.config.ts \
+          --replace-fail "const commitHash = execSync('git rev-parse --verify --short HEAD').toString().trim();" 'const commitHash = "";' \
+          --replace-fail "const versionTag = execSync('git --no-pager tag --sort -taggerdate --points-at HEAD')" 'const versionTag = "v${version}";' \
+          --replace-fail "const gitClean = execSync('git status --porcelain').toString() ? false : true;" 'const gitClean = true;'
```

Comment:
> wouldnt it make this easier to read in a patch file?

I originally did that, because the version changes with each version upgrade.
When I did my packaging in https://github.com/C0D3-M4513R/slimevr-nixos/ I was doing replacements like this, so I can more rely on nixos's automated upgrading, without having to edit anything manually (since I was developing with the intent of pr'ing to nix eventually)

Reply:
Oops, wrong account.

Reply:
Also the current patchfile will completely remove the version, which will probably remove the version display on the top-left, right?
![image](https://github.com/user-attachments/assets/072c075f-470d-42a9-be4e-ae6fb8c91420)

Reply:
The version's still being replaced after the patch file.